.grit/patterns/jest_to_vitest.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Jest to Vitest
+---
+
+Convert Jest test to Vitest.
+
+tags: #migration, #js
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern updated_imports() {
+  call_expression($function) where {
+    $function <: or {
+      js"test",
+      js"expect",
+      js"describe",
+      js"it",
+      js"beforeEach",
+      js"afterEach",
+      js"beforeAll",
+      js"afterAll"
+    } where {
+      $source = `'vitest'`,
+      $function <: ensure_import_from($source),
+      $vi = `vi`,
+      $vi <: ensure_import_from($source)
+    }
+  }
+}
+
+or {
+  // Imports
+  updated_imports(),
+  `import $_ from 'jest'` => .,
+
+  `jest.requireActual($module)` => `await vi.importActual($module)`,
+
+  `JEST_WORKER_ID` => `VITEST_POOL_ID`,
+
+  // module mocks
+  `jest.mock($module, $mockImplementation)` where { 
+    $mockImplementation <: arrow_function($body) where { 
+      $body <: literal_value() => `({ 
+        default: $body 
+      })`
+    }
+  } => `vi.mock($module, $mockImplementation)`,
+  `jest.mock` => `vi.mock`,
+
+  // done callback
+  `$sym($description, $callback)` where {
+    and {
+        $sym <: contains {
+            or { `it`, `test`}
+        },
```

Comment:
`contains` is somewhat expensive`. Why not just use the `or` directly?

Reply:
Thanks @morgante Sorry I didn't noticed this before. I updated them to use `or` directly